R/irace_helpers.R

```diff
@@ -0,0 +1,64 @@
+paradoxToIrace = function(ps){
+  assertClass(ps, "ParamSet")
+  # what about ParamUty = vector numeric/real
+  class_lookup = data.frame(paradox = c("ParamLgl","ParamInt","ParamDbl","ParamFct"),
+                            irace = c("c","i","r","c"), stringsAsFactors = FALSE)
+  name <- ps$ids()
+  type <- class_lookup[match(ps$class, class_lookup$paradox), 2]
+  range <- getIraceRange(ps)
+  if(ps$has_deps){
+    condition <- getIraceCondition(ps)
+  } else {
+    condition = NULL
+  }
+
+  # does mlr3tuning deal with trafo internally or passed to irace? irace only handle log
+  par.tab <- paste(name, '""', type, range, condition, collapse = "\n")
+  irace::readParameters(text = par.tab)
+}
+getIraceRange = function(ps){
+  rng = data.table(lower = ps$lower, upper = ps$upper, lvl = ps$levels)
+
+  apply(rng, 1, function(x){
+    if(is.na(x[[1]]))
+      return(paste0("(",paste0(x[[3]], collapse = ","),")"))
+    else
+      return(paste0("(",x[[1]],",",x[[2]],")"))
+  })
+}
+getIraceCondition = function(ps){
+  cond = apply(ps$deps, 1, function(x){
+    if(class(x[[3]])[1] == "CondEqual"){
+      condition = paste("|",x[[2]], "==", x[[3]]$rhs)
```

Comment:
It has been tested. irace takes in everything in an expression (or deparsed?) form, so all numerics/non-numerics look identical but it then internally coerces back to required form

Reply:
> `||` instead of `|` would be nicer here

Where do you want `||` ? The use of `|` in the paste in line 32 is required by `irace`. It isn't stating "or" but that the condition is optional.

Reply:
> also has this been tested? it looks like `$rhs` should be quoted if it is non-numeric.

You were right, this is now fixed

Reply:
ah ok didn't know about the `"|"`